### DISCUSSION TITLE
values unavailable ?

### DISCUSSION THREAD

**Discussion by nigelspowage** [macxq/foxess-ha#37] [General]
I installed this integration last night and everything appeared to work. This morning all the values are appearing as unavailable.
![image](https://user-images.githubusercontent.com/34656171/151933173-83a24f4a-a7ea-4457-89a0-8873a85be632.png)

Any idea why and how to fix it ?

**Comment by macxq** [comment]
@nigelspowage what can you find in logs ?

**Comment by nigelspowage** [comment]
I've just found the following in the logs:

`This error originated from a custom integration.

Logger: custom_components.foxess.sensor
Source: custom_components/foxess/sensor.py:153
Integration: foxess (documentation, issues)
First occurred: 08:37:58 (1 occurrences)
Last logged: 08:37:58

Unexpected error fetching FoxESS data: Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 187, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 147, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/foxess/sensor.py", line 105, in async_update_data
    token = await authAndgetToken(hass, username, hashedPassword)
  File "/config/custom_components/foxess/sensor.py", line 153, in authAndgetToken
    response = json.loads(restAuth.data)
  File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
`

and 

`This error originated from a custom integration.

Logger: custom_components.foxess.sensor
Source: custom_components/foxess/sensor.py:130
Integration: foxess (documentation, issues)
First occurred: 08:37:58 (1 occurrences)
Last logged: 08:37:58

FoxESS Cloud initializaction failed, fix error and restar ha`

**Comment by macxq** [comment]
update to new release - it is the user agent problem

**Comment by nigelspowage** [comment]
Thanks, will do :)

**Comment by nigelspowage** [comment]
It's back, thank you :)

**Comment by williamjeccles** [comment]
Update also resolved this for me, seems Foxess are playing games. They simply need to scale their platform instead of blocking legitimate users of it.

**Comment by williamjeccles** [comment]
Sadly appears foxess is down again but this time their end.

**Comment by BigManDave** [comment]
We've been IP blocked. See #28
